src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -304,6 +304,13 @@ private BuildEventArgs ReadTargetSkippedEventArgs()
                 condition = ReadOptionalString();
                 evaluatedCondition = ReadOptionalString();
                 originallySucceeded = ReadBoolean();
+
+                // Attempt to infer skip reason from the data we have
+                skipReason = condition != null ?
+                    TargetSkipReason.ConditionWasFalse
+                    : originallySucceeded ?
+                        TargetSkipReason.PreviouslyBuiltSuccessfully
+                        : TargetSkipReason.PreviouslyBuiltUnsuccessfully;
```

Comment:
```suggestion
                skipReason = condition != null ?
                    TargetSkipReason.ConditionWasFalse // condition expression only stored when false
                    : originallySucceeded ?
                        TargetSkipReason.PreviouslyBuiltSuccessfully
                        : TargetSkipReason.PreviouslyBuiltUnsuccessfully;
```

? I had basically the same questions so I'd like a permanent comment.

Reply:
Actually, as an alternative, maybe change it to:
skipReason = condition == false ?
?
That should be equally valid but far less surprising.